python/lancedb/remote/client.py

```diff
@@ -151,10 +151,13 @@ async def post(
             return await deserialize(resp)
 
     @_check_not_closed
-    async def list_tables(self):
+    async def list_tables(self, limit: int, page_token: str):
         """List all tables in the database."""
-        json = await self.get("/v1/table/", {})
-        return json["tables"]
+        try:
+            json = await self.get("/v1/table/", {"limit": limit, "page_token": page_token})
+            return False, json["tables"]
```

Comment:
we can just determine if we are done by an empty `json["table"]`. if the returned data is empty there is no more to list